src/Sylius/Bundle/UserBundle/EventListener/UserDeleteListener.php

```diff
@@ -56,4 +55,18 @@ public function deleteUser(ResourceControllerEvent $event): void
             $flashBag->add('error', 'Cannot remove currently logged in user.');
         }
     }
+
+    private function isTryingToDeleteLoggedInAdminUser(UserInterface $user): bool
+    {
+        $isTryingToDeleteLoggedInUser = false;
+        if ($user instanceof AdminUserInterface) {
+            $token = $this->tokenStorage->getToken();
+            if ($token) {
+                $loggedUser = $token->getUser();
+                $isTryingToDeleteLoggedInUser = ($loggedUser instanceof AdminUserInterface) && ($loggedUser->getId() === $user->getId());
```

Comment:
Hmm... right now I'm wondering, would it even work? `$loggedUser` would always be `AdminUserInterface`, as we perform delete only from admin panel. We should probably instead check if `$user` is an admin because if it is **and** its ID is the same as logged user id, then deletion should be broken.

Reply:
I removed the test checking that logged user is `AdminUserInterface`
I check now that `$user` is an admin by checking that he has the role `ROLE_ADMINISTRATION_ACCESS`

For writing a scenario in this case, I am not sure that he can easily be handled with behat because we need to force IDs of Admin User and Customer in database.
That's why I tested this change with phpspec